### PR TITLE
[JENKINS-62290] Make Email Extension Plugin accesible to Jenkins.MANAGE

### PR DESCRIPTION
**Pull request by escoem**
[JENKINS-62290](https://issues.jenkins-ci.org/browse/JENKINS-62290)

This is using the new `Jenkins.MANAGE` permission. It is intended to grant an intermediate access level to global configuration for non admins but more than read users.

The groovy scripts shouldn't be a problem because of the script-security usage.

(I've also updated the the minimum Jenkins core LTS including the `Jenkins.MANAGE` and updated to a newer parent pom).

### CODE REVIEW

File: src/test/java/hudson/plugins/emailext/ExtendedEmailPublisherTest.java [hunk 1]
```diff
@@ -124,7 +124,7 @@ private void setUpSecurity() {
                                           // otherwise we would need to create users for each email address tested, to bypass SECURITY-372 fix:
                                           .grant(Jenkins.READ, Item.READ).everywhere().toAuthenticated()
                                           // TODO I had plans for tests where bob would approve scripts written by alice
-                                          .grant(Jenkins.ADMINISTER).everywhere().to("bob")
+                                          .grant(Jenkins.MANAGE).everywhere().to("bob")
```
Comment (amuniz) [inline src/test/java/hudson/plugins/emailext/ExtendedEmailPublisherTest.java:127]: I think this tests should stay as they were (using `Jenkins.ADMINISTER`) which should still pass. And create a specific test for `Jenkins.MANAGE` (or an additional assert in an existing one).

**Comment by amuniz** [comment]
Build is failing with a compilation error:
```
[ERROR] /home/jenkins/workspace/Plugins_email-ext-plugin_PR-208/src/main/java/hudson/plugins/emailext/ExtendedEmailPublisherDescriptor.java:[821,23] cannot find symbol
symbol:   variable MANAGE
location: class jenkins.model.Jenkins
```
Because it's running with `-Djenkins.version=2.164.1`, not sure why (a bug in the pipeline?)

File: src/test/java/hudson/plugins/emailext/ExtendedEmailPublisherTest.java (hunk 1 above)
Comment (slide) [reply src/test/java/hudson/plugins/emailext/ExtendedEmailPublisherTest.java:127]: Agreed

File: src/test/java/hudson/plugins/emailext/plugins/content/ScriptContentSecureTest.java [hunk 2]
```diff
@@ -47,7 +47,7 @@ public void setup() {
         j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
 
         j.jenkins.setAuthorizationStrategy(new MockAuthorizationStrategy()
-                                                   .grant(Jenkins.ADMINISTER).everywhere().to("bob")
+                                                   .grant(Jenkins.MANAGE).everywhere().to("bob")
```
Comment (amuniz) [inline src/test/java/hudson/plugins/emailext/plugins/content/ScriptContentSecureTest.java:50]: Revert this?

File: src/test/java/hudson/plugins/emailext/plugins/trigger/AbstractScriptTriggerTest.java [hunk 3]
```diff
@@ -43,7 +43,7 @@ public void setup() {
         j.jenkins.setAuthorizationStrategy(new MockAuthorizationStrategy()
                 // otherwise we would need to create users for each email address tested, to bypass SECURITY-372 fix:
                 .grant(Jenkins.READ, Item.READ, Item.DISCOVER, Item.CONFIGURE).everywhere().toAuthenticated()
-                .grant(Jenkins.ADMINISTER).everywhere().to("bob"));
+                .grant(Jenkins.MANAGE).everywhere().to("bob"));
```
Comment (amuniz) [inline src/test/java/hudson/plugins/emailext/plugins/trigger/AbstractScriptTriggerTest.java:46]: Revert this?

**Comment by slide** [comment]
> Build is failing with a compilation error:
> 
> ```
> [ERROR] /home/jenkins/workspace/Plugins_email-ext-plugin_PR-208/src/main/java/hudson/plugins/emailext/ExtendedEmailPublisherDescriptor.java:[821,23] cannot find symbol
> symbol:   variable MANAGE
> location: class jenkins.model.Jenkins
> ```
> 
> Because it's running with `-Djenkins.version=2.164.1`, not sure why (a bug in the pipeline?)

That's part of the recommendedConfigurations() in buildPlugin. https://github.com/jenkins-infra/pipeline-library/blob/master/vars/buildPlugin.groovy#L258

**Comment by escoem** [comment]
Hi @slide I have addressed your requests (new test and restore the three other). Regarding 2.222.1, that LTS is required to build this PR.

**Comment by slide** [comment]
Well, we may not be able to merge this for a while then. I need to review the plugin installation stats and see how many people are still on 2.164.x.

**Comment by escoem** [comment]
Hi @slide I have used reflection to avoid core bump

**Pull request merged**